airflow/providers/yandex/operators/yandexcloud_dataproc_operator.py

```diff
@@ -0,0 +1,334 @@
+    """Runs Mapreduce job in Data Proc cluster."""
+
+    # pylint: disable=too-many-arguments
+
+    @apply_defaults
+    def __init__(self,
+                 main_class=None,
+                 main_jar_file_uri=None,
+                 jar_file_uris=None,
+                 archive_uris=None,
+                 file_uris=None,
+                 args=None,
+                 properties=None,
+                 name='Mapreduce job',
+                 cluster_id=None,
+                 connection_id=None,
+                 *arguments,
+                 **kwargs):
+        super().__init__(*arguments, **kwargs)
+        self.main_class = main_class
+        self.main_jar_file_uri = main_jar_file_uri
+        self.jar_file_uris = jar_file_uris
+        self.archive_uris = archive_uris
+        self.file_uris = file_uris
+        self.args = args
+        self.properties = properties
+        self.name = name
+        self.cluster_id = cluster_id
+        self.connection_id = connection_id
+
+    def execute(self, context):
+        cluster_id = self.cluster_id or context['task_instance'].xcom_pull(key='cluster_id')
+        connection_id = self.connection_id or context['task_instance'].xcom_pull(
+            key='yandexcloud_connection_id'
+        )
+        self.hook = DataprocHook(
+            connection_id=connection_id,
+        )
+        self.hook.client.create_mapreduce_job(
+            main_class=self.main_class,
+            main_jar_file_uri=self.main_jar_file_uri,
+            jar_file_uris=self.jar_file_uris,
+            archive_uris=self.archive_uris,
+            file_uris=self.file_uris,
+            args=self.args,
+            properties=self.properties,
+            name=self.name,
+            cluster_id=cluster_id,
+        )
+
+
+class DataprocCreateSparkJobOperator(YandexCloudBaseOperator):
+    """Runs Spark job in Data Proc cluster."""
+
+    # pylint: disable=too-many-arguments
+
+    @apply_defaults
+    def __init__(self,
+                 main_class=None,
+                 main_jar_file_uri=None,
+                 jar_file_uris=None,
+                 archive_uris=None,
+                 file_uris=None,
+                 args=None,
+                 properties=None,
+                 name='Spark job',
+                 cluster_id=None,
+                 connection_id=None,
+                 *arguments,
+                 **kwargs):
+        super().__init__(*arguments, **kwargs)
+        self.main_class = main_class
+        self.main_jar_file_uri = main_jar_file_uri
+        self.jar_file_uris = jar_file_uris
+        self.archive_uris = archive_uris
+        self.file_uris = file_uris
+        self.args = args
+        self.properties = properties
+        self.name = name
+        self.cluster_id = cluster_id
+        self.connection_id = connection_id
+
+    def execute(self, context):
+        cluster_id = self.cluster_id or context['task_instance'].xcom_pull(key='cluster_id')
```

Comment:
I think it would be better to add cluster_id to templated fields. Then users can retrieve this value from any key by passing `cluster_id="{{ task_instance.xcom_pull('other_task_id', key='custom_key') }}"` :)

Reply:
Done